fix python tests on julia v1.12

bjarthur opened the pull request:
CondaPkg.toml now needs to be in the root dir not test/.  with this PR, https://github.com/JuliaIO/Zarr.jl/pull/216 and https://github.com/JuliaIO/Zarr.jl/pull/218 should pass.

mkitti commented:
Please add tests for to run on `"lts"`:
https://github.com/julia-actions/setup-julia?tab=readme-ov-file#examples

nhz2 commented:
Putting CondaPkg.toml at the root means that PythonCall.jl may try to install those packages if someone is using Zarr.jl, so I think it needs to stay in the test directory.

bjarthur commented:
good catch @nhz2 !  i deleted the CondaPkg.toml file entirely and now using CondaPkg.add in test/python.jl to add the zarr and numpy deps.

have also updated CI.yml to test on LTS.

ready to merge?
